scripts/build_arrow_linux.sh

```diff
@@ -0,0 +1,123 @@
+#!/bin/bash -e
+#
+# build arrow on linux
+#
+function usage() {
+    echo "Usage: `basename $0` [-t <Release|Debug>]"
+    echo "Build ARROW"
+    echo "-t <Release/Debug> : Release or Debug builds"
+    exit 2
+}
+set -o pipefail
+
+ARROW_VERSION=0.17.1
+
+#If its not for XP use gcc52
+if [[ -z "$XP_BUILD" ]] ; then
+  export CC="/usr/lib64/ccache/gcc52"
+  export CXX="/usr/lib64/ccache/g++52"
+else
+  export CC="gcc82"
+  export CXX="g++82"
+fi
+
+DIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" && pwd )"
+source $DIR/_init.sh $@
+source $DIR/utils.sh
+
+[[ -n "$GET_VERSION" ]] && echo $AZURE_VERSION && exit 0
+
+ARROW_SOURCE_DIR=$DEPS_DIR/arrow
+ARROW_BUILD_DIR=$DEPENDENCY_DIR/arrow
+ARROW_DEPS_BUILD_DIR=$DEPENDENCY_DIR/arrow_deps
+ARROW_CMAKE_BUILD_DIR=$ARROW_SOURCE_DIR/cpp/cmake-build
+
+GIT_REPO="https://github.com/apache/arrow.git"
```

Comment:
Not sure if we should fetch arrow directly from here, it is not a small fetch. Let me check if we should use some cache here.